unstructured/documents/elements.py

```diff
@@ -167,6 +167,9 @@ class ElementMetadata:
     # Metadata extracted via regex
     regex_metadata: Optional[Dict[str, List[RegexMetadata]]] = None
 
+    # Title depth to map chipper elements Headline and Subheadline
+    title_depth: Optional[int] = None
```

Comment:
design wise it seems we are throwing everything into the `ElementMetaData` and results in a lot of `Optional` fields. This can be cumbersome down the road when this list becomes unwieldily long. Maybe we should consider refactor this into a base class and mixins/inherited classes for things like `TitleMetaData`. The optional also creates uncertainty in downstream process when we are not guaranteed an input `ElementMetaData` will have a certain attribute for processing and we'd have to check if a field is `None` potentially a lot. E.g., a downstream process function on metadata would always need an accompanying parent `Element`'s category and knowledge from dev what category has which attributes.

Reply:
makes sense, in that case we would have to refactor several `Optional` metadata fields in `ElementMetadata` and consider a similar logic + grouping. Totally in favour of having a shorter list of Optional fields if any.

Reply:
Other elements may benefit from having a depth such as bullet/list items. Could we rename this parameter to simply `depth` to enable it to be used across element types? Also, I think a sensible default would be 0, as in no depth

Reply:
After discussion, `category_depth` will better represent the intention of the field -- the depth relative to the element's category (e.g. Title, Bullet, or List) not the depth relative to all elements